cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/ClusterBootstrap.scala

```diff
@@ -108,8 +109,19 @@ final class ClusterBootstrap(implicit system: ExtendedActorSystem) extends Exten
     _selfContactPointUri.success(baseUri)
 
   /** INTERNAL API */
-  @InternalApi private[akka] def selfContactPoint: Future[Uri] =
-    _selfContactPointUri.future
+  @InternalApi private[akka] def selfContactPoints: Future[immutable.Seq[(String, Int)]] =
+    _selfContactPointUri.future.map { uri =>
+      if (system.settings.config.hasPath("akka.discovery.kubernetes-api.pod-domain") && uri.authority.host.isIPv4()) {
```

Comment:
Somewhat ugly that we have to have special case for kubernetes here. Wonder if it could be defined or set by the kubernetes-api "plugin" instead. Perhaps we should revisit the design of how the `_selfContactPointUri` is set in a separate ticket.

Reply:
I agree. I didn't like having k8s code 'slip in' here, but making this 'pluggable' as well seemed a bit over-engineered. Indeed we might find a nice way to do it later, though.